SwiftCompilerSources/Sources/Optimizer/Utilities/OptUtils.swift

```diff
@@ -539,6 +560,63 @@ extension Function {
     }
     return nil
   }
+
+  var mayBindDynamicSelf: Bool {
+    self.bridged.mayBindDynamicSelf()
+  }
+
+  enum NonFailureExitBBsSearchResult {
+    case allBlocksNotUnderstood
+    case allBlocksUnderstood([BasicBlock])
+  }
+
+  func findAllNonFailureExitBBs() -> NonFailureExitBBsSearchResult {
```

Comment:
We can do better than the C++ implementation in swift.
Instead of returning an array of result blocks, you can just define a property in the BasicBlock: `var isNonFailureExitBlock: Bool`. If you want to get all of them, you can just write `function.blocks.filter(\.isNonFailureExitBlock)`.